### ISSUE TITLE
How to train small and large bboxes at the same time?

### ISSUE THREAD

**Issue by hshintaro** [ultralytics/ultralytics#15912]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained two objects where the box sizes differ by a factor of 30 times.
As a result, the large box could be learned, but the small box could not.
I thought that the larger box would be prioritized due to the design of the loss function, so I added focal loss as follows, but it didn't solve the problem.

Please advise how to resolve this issue.

```
# Parameters
nc: 2 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolov10n.yaml' will call yolov10.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024]

backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, SCDown, [512, 3, 2]] # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, SCDown, [1024, 3, 2]] # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9
  - [-1, 1, PSA, [1024]] # 10

# YOLOv10.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, C2f, [512]] # 13

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, C2f, [256]] # 16 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 13], 1, Concat, [1]] # cat head P4
  - [-1, 3, C2f, [512]] # 19 (P4/16-medium)

  - [-1, 1, SCDown, [512, 3, 2]]
  - [[-1, 10], 1, Concat, [1]] # cat head P5
  - [-1, 3, C2fCIB, [1024, True, True]] # 22 (P5/32-large)

  - [[16, 19, 22], 1, v10Detect, [nc]] # Detect(P3, P4, P5)


anchors:
  - [5,5, 10,10, 20,20]  # Small-scale anchors
  - [150,200, 200, 150, 150,150]  # Medium-scale anchors
  - [500,200, 200,500, 500,500]  # Large-scale anchors

loss: focal_loss
```

**Comment by glenn-jocher** [comment]
@hshintaro to train both small and large bounding boxes effectively, consider adjusting the anchor sizes to better match the scale of your objects and ensure your dataset is well-balanced. Additionally, you might want to experiment with different loss functions or multi-scale training techniques. For more detailed guidance, please refer to our documentation at https://docs.ultralytics.com/guides/yolo-common-issues/.

**Comment by Y-T-G** [comment]
You can try the YOLOv8n-p2 model.

`model = YOLO("yolov8n-p2.yaml")`

**Comment by hshintaro** [comment]
Thank you for your reply.
This problem was solved by introducing the p1/p2 network, but increasing the resolution caused problems with machine resources.

**Comment by glenn-jocher** [comment]
You're welcome! To manage resource usage with higher resolutions, consider optimizing your batch size or using a more powerful GPU. If you need further assistance, feel free to ask.
